src/connector/src/sink/big_query.rs

```diff
@@ -280,10 +309,15 @@ pub struct BigQuerySinkWriter {
     pub config: BigQueryConfig,
     schema: Schema,
     pk_indices: Vec<usize>,
-    client: Client,
+    client: StorageWriterClient,
     is_append_only: bool,
-    insert_request: TableDataInsertAllRequest,
-    row_encoder: JsonEncoder,
+    row_encoder: ProtoEncoder,
+    writer_pb_schema: ProtoSchema,
+    message_descriptor: MessageDescriptor,
+    write_stream: String,
+    proto_field: Option<FieldDescriptor>,
```

Comment:
It's fine to store the field also for append only case and not use Option, since the field descriptor is light weight.